sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java

```diff
@@ -48,12 +53,26 @@ public String toString() {
   public static MetricName named(String namespace, String name) {
     checkArgument(!Strings.isNullOrEmpty(namespace), "Metric namespace must be non-empty");
     checkArgument(!Strings.isNullOrEmpty(name), "Metric name must be non-empty");
-    return new AutoValue_MetricName(namespace, name);
+    return new AutoValue_MetricName(namespace, name, ImmutableMap.of());
+  }
+
+  public static MetricName named(
+      String namespace, String name, ImmutableMap<String, String> labels) {
```

Comment:
a possible fix for the vendor issue would be to change this back to map and do
ImmutableMap.copyOf when passing to AutoValue.

This might be nicer api since it makes tests easier etc. If callers use ImmutableMap there is no performance overhead for the copy.